### ISSUE TITLE
I don't see a value for the first column for targets in dataloader.py

### ISSUE THREAD

**Issue by shuiyihang** [WongKinYiu/yolov9#521]
in loss_tal_dual.py 151 lines，This code uses the first column
     i = targets[:, 0]  # image index

but in dataloader.py 713 lines：
labels_out = torch.zeros((nl, 6))
        if nl:
            labels_out[:, 1:] = torch.from_numpy(labels)

I don't see a value for the first column for targets in dataloader.py
